iroh-bytes/src/downloader.rs

```diff
@@ -454,540 +503,495 @@ struct Service<G: Getter, D: Dialer> {
     concurrency_limits: ConcurrencyLimits,
     /// Channel to receive messages from the service's handle.
     msg_rx: mpsc::Receiver<Message>,
-    /// Peers available to use and their relevant information.
+    /// Active connections
     nodes: HashMap<NodeId, ConnectionInfo<D::Connection>>,
     /// Queue to manage dropping nodes.
     goodbye_nodes_queue: delay_queue::DelayQueue<NodeId>,
-    /// Requests performed for download intents. Two download requests can produce the same
-    /// request. This map allows deduplication of efforts.
-    current_requests: HashMap<DownloadKind, ActiveRequestInfo>,
-    /// Downloads underway.
-    in_progress_downloads: JoinSet<DownloadRes>,
-    /// Requests scheduled to be downloaded at a later time.
-    scheduled_requests: HashMap<DownloadKind, PendingRequestInfo>,
-    /// Queue of scheduled requests.
-    scheduled_request_queue: delay_queue::DelayQueue<DownloadKind>,
+    /// Queue of pending downloads.
+    queue: LinkedHashSet<DownloadKind>,
+    /// Information about pending and active requests
+    requests: HashMap<DownloadKind, RequestInfo>,
+    /// State of running downloads
+    active_requests: HashMap<DownloadKind, ActiveRequestInfo>,
+    /// Tasks for currently running transfers.
+    in_progress_downloads: JoinSet<(DownloadKind, InternalDownloadResult)>,
+    /// Progress tracker
+    progress_tracker: ProgressTracker,
+    /// Registered intents with progress senders and result callbacks.
+    intents: HashMap<IntentId, IntentCallbacks>,
```

Comment:
Why not store this in `RequestInfo`? if the intents are already being hashed there

Reply:
Agreed and changed!
This was a bit different in an earlier design, now it makes sense to inline this in the request info. Also simplifies the code and removes some lines.